model/project_ref.go

```diff
@@ -3014,21 +3014,42 @@ func (p *ProjectRef) GetProjectSetupCommands(opts apimodels.WorkstationSetupComm
 	return cmds, nil
 }
 
+func updateNextRunTime(baseTime time.Time, definition *PeriodicBuildDefinition) (time.Time, error) {
+	var nextRunTime time.Time
+	var err error
+	if definition.IntervalHours > 0 {
+		nextRunTime = baseTime.Add(time.Duration(definition.IntervalHours) * time.Hour)
+	} else {
+		nextRunTime, err = GetNextCronTime(baseTime, definition.Cron)
+		if err != nil {
+			return time.Time{}, errors.Wrap(err, "getting next run time with cron")
+		}
+	}
+	return nextRunTime, nil
+}
+
 // UpdateNextPeriodicBuild updates the periodic build run time for either the project
 // or repo ref depending on where it's defined.
 func UpdateNextPeriodicBuild(ctx context.Context, projectId string, definition *PeriodicBuildDefinition) error {
-	var nextRunTime time.Time
-	var err error
 	baseTime := definition.NextRunTime
 	if utility.IsZeroTime(baseTime) {
 		baseTime = time.Now()
 	}
-	if definition.IntervalHours > 0 {
-		nextRunTime = baseTime.Add(time.Duration(definition.IntervalHours) * time.Hour)
-	} else {
-		nextRunTime, err = GetNextCronTime(baseTime, definition.Cron)
+	nextRunTime, err := updateNextRunTime(baseTime, definition)
+	if err != nil {
+		return errors.Wrap(err, "updating next run time")
+	}
+	now := time.Now()
+	// If the nextRunTime is still in the past, bring its base time up to present and re-calculate it.
```

Comment:
Could we clarify that this could happen if the periodic build's pre-existing next run time is in the past?